internal/webhooks/capp_validation_utils.go

```diff
@@ -46,24 +47,37 @@ func getManagedClusters(r client.Client, ctx context.Context) ([]string, error)
 
 // validateDomainName checks if the hostname is valid domain name and not part of the cluster's domain.
 // it returns aggregated error if any of the validations falied.
-func validateDomainName(domainname string) (errs *apis.FieldError) {
-	if domainname == "" {
+func validateDomainName(domainName string) (errs *apis.FieldError) {
+	if domainName == "" {
 		return nil
 	}
-	err := validation.IsFullyQualifiedDomainName(field.NewPath("name"), domainname)
+	err := validation.IsFullyQualifiedDomainName(field.NewPath("name"), domainName)
 	if err != nil {
 		errs = errs.Also(apis.ErrGeneric(fmt.Sprintf(
-			"invalid name %q: %s", domainname, err.ToAggregate()), "name"))
+			"invalid name %q: %s", domainName, err.ToAggregate()), "name"))
 	}
 
 	clusterLocalDomain := network.GetClusterDomainName()
-	if strings.HasSuffix(domainname, "."+clusterLocalDomain) {
+	if strings.HasSuffix(domainName, "."+clusterLocalDomain) {
 		errs = errs.Also(apis.ErrGeneric(
-			fmt.Sprintf("invalid name %q: must not be a subdomain of cluster local domain %q", domainname, clusterLocalDomain), "name"))
+			fmt.Sprintf("invalid name %q: must not be a subdomain of cluster local domain %q", domainName, clusterLocalDomain), "name"))
+	}
+	if domainNameTaken := isDomainNameTaken(domainName); domainNameTaken {
+		errs = errs.Also(apis.ErrGeneric(
+			fmt.Sprintf("invalid name %q: must not be already in use", domainName), "name"))
+
 	}
 	return errs
 }
 
+// isDomainNameTaken checks if the given hostname is already in use.
+func isDomainNameTaken(domainName string) bool {
+	if _, err := net.LookupHost(domainName); err == nil {
+		return true
+	}
+	return false
+}
```

Comment:
Is there a way to be more specific about the error? What if the record does exist but there's been an error connecting to the DNS server? That would reflect as `false` here.

Reply:
But then we would have 31 tests instead of a nice, round 30 :(